### PR TITLE
[RLlib] Metrics do-over 03: Switch over `Learner` to new MetricsLogger API.

### PR DESCRIPTION
**Pull request by sven1977**
Metrics do-over 03: Switch over `Learner` to new MetricsLogger API.
* Retire Learner's current `register_metric(s)` APIs and replace with the new MetricsLogger API, which is also already used by EnvRunners AND Algorithm (if the new stack is activated).
* TODO: We have to find a good way of transitioning those users that are currently on the hybrid stack and are therefore NOT using MetricsLogger on Algo yet (how to we reduce over the LearnerGroup returned reduced Stats-dicts). Possible solution: Have LearnerGroup act as an interceptor in case enable_rl_module_and_learner is True, but enable_env_runner_and_connector_v2 is False (hybrid stack). In this case, LearnerGroup will reduce the results on behalf of Algorithm and return a simple (non-Stats) dict, which hybrid-stack Algos know how to handle.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/appo/tf/appo_tf_learner.py [hunk 1]
```diff
@@ -160,8 +160,7 @@ def compute_loss_for_module(
         )
 
         # Register important loss stats.
-        self.register_metrics(
-            module_id,
+        self.metrics.log_dict(
```
Comment (simonsays1980) [inline rllib/algorithms/appo/tf/appo_tf_learner.py:163]: Nice. This is now much cleaner. Learner only learns and metrics logger logs.

File: rllib/algorithms/dqn/torch/dqn_rainbow_torch_learner.py [hunk 2]
```diff
@@ -190,22 +190,33 @@ def compute_loss_for_module(
                 * loss_fn(reduction="none")(q_selected, q_selected_target)
             )
 
-        self.register_metrics(
-            module_id,
+        # Log the TD-error with reduce=None, such that - in case we have n parallel
+        # Learners - we will re-concatenate the produced TD-error tensors to yield
+        # a 1:1 representation of the original batch.
+        self.metrics.log_value(
```
Comment (simonsays1980) [inline rllib/algorithms/dqn/torch/dqn_rainbow_torch_learner.py:196]: Awesome!

File: rllib/core/learner/learner_group.py [hunk 3]
```diff
@@ -208,14 +219,13 @@ def update_from_batch(
...
-        ),
-        # TODO (sven): Deprecate the following args. They should be extracted from the
-        #  LearnerHyperparameters of those specific algorithms that actually require
-        #  these settings.
+        # TODO (sven): Deprecate the following args. They should be extracted from
+        #  self.config of those specific algorithms that actually require these
+        #  settings.
         minibatch_size: Optional[int] = None,
         num_iters: int = 1,
+        # Already deprecated args.
```
Comment (simonsays1980) [inline rllib/core/learner/learner_group.py:227]: Awesome. This reduces a lot of complexity for users. This reduce function was hard to develop w/o having the result dict itself.

**Review by simonsays1980 — APPROVED** [review]
LGTM. Reduces a lot of complexity in logging.

File: rllib/core/learner/learner_group.py (hunk 3 above)
Comment (sven1977) [reply rllib/core/learner/learner_group.py:227]: Yes, totally. It was very messy. Also, it wasn't very clear, where you should reduce, in the Learner (after n SGD iters), in the LearnerGroup (after receiving m Learner results), on the Algo side?

**Review by aslonnie — APPROVED** [review]
approval for docs.

**Pull request merged**